modules/biobank/php/pool.class.inc

```diff
@@ -0,0 +1,266 @@
+<?php declare(strict_types=1);
+/**
+ * Pool Class
+ * This class contains an instance of a database handling that is needed to
+ * permanently store and retrieve Pool Object instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Entity
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+namespace LORIS\biobank;
+
+/**
+ * Pool Class
+ * This class contains an instance of a database handling that is needed to
+ * permanently store and retrieve Pool Object instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Entity
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class Pool implements \JsonSerializable, \LORIS\Data\DataInstance
+{
+    /**
+     * Persistent Instance variables.
+     *
+     * These variables are directly mapped to the fields of the database schema
+     * that describe a pool entity.
+     *
+     * @access private
+     */
+    private $id;          //int
+    private $label;       //string
+    private $specimenIds; //array
+    private $candidateId; //int
+    private $sessionId;   //int
+    private $centerId;    //int
+    private $date;        //string
+    private $time;        //string
+
+    /**
+     * Initiliazes a new instance of the Pool Class
+     */
+    function __construct()
+    {
+    }
+
+    /**
+     * Sets the pool's ID
+     *
+     * @param int $id the pool's ID
+     *
+     * @return void
+     */
+    public function setId(?int $id = null) : void
```

Comment:
Should this be allowed to be null? Seems important

Reply:
Yeah, unfortunately I don't really see a way around this. It won't receive an ID until it is inserted into the database, so any new `Pool` that is being created won't have an ID. Alternatively, a check can be done before setting an ID to make sure that it exists before setting it, but I think functionally that results in the same outcome.

@ridz1208 thoughts?

Reply:
I think this is fine, the object having a nullable ID does not mean that the SQL entry can have one, as long as we properly persist it. I dont see the problem